### ISSUE TITLE
Beat process stuck at "Acquiring lock"

### ISSUE THREAD

**Issue by richardasaurus** [sibson/redbeat#85]
Very often this is happening with my celery beat process. Any idea why?

I'm using redbeat scheduler, sqs broker. It stays like this for a long time, 10mins+.
The rest of my application can connect to and use redis quickly and constantly.

```
[2018-03-08 19:35:39,341: DEBUG/MainProcess] Setting default socket timeout to 30
[2018-03-08 19:35:39,341: INFO/MainProcess] beat: Starting...
[2018-03-08 19:35:39,412: DEBUG/MainProcess] Stored entry: <RedBeatSchedulerEntry: periodic-heartbeat ai_platform.tasks.task_heartbeat() <freq: 25.00 seconds>
[2018-03-08 19:35:39,414: DEBUG/MainProcess] Stored entry: <RedBeatSchedulerEntry: task-daily-forecast ai_platform.tasks.task_xxx() <crontab: 20 19 * * * (m/h/d/dM/MY)>
[2018-03-08 19:35:39,415: DEBUG/MainProcess] beat: Ticking with max interval->5.00 seconds
[2018-03-08 19:35:39,415: DEBUG/MainProcess] beat: Acquiring lock...
```

```
celery==4.1.0
celery-redbeat==0.11.1
```

**Comment by pdoggi** [comment]
I'm experiencing the same thing on EC2.  
```
[2018-03-12 18:30:38,423: DEBUG/MainProcess] beat: Acquiring lock...
[2018-03-12 18:43:09,712: DEBUG/MainProcess] Setting default socket timeout to 30
```

```
```
On my dev machine.

**Comment by richardasaurus** [comment]
"fixed" by adding the following to celery config module in my project:

```
CELERYBEAT_MAX_LOOP_INTERVAL = 10
REDBEAT_LOCK_TIMEOUT = CELERYBEAT_MAX_LOOP_INTERVAL * 5
```

**Comment by pdoggi** [comment]
Thanks for sharing, @richardasaurus. 

My problem was that my script to restart the beat thread doesn't kill the previous beat thread cleanly and would leave a lock in redis. When I `DEL redbeat::lock`, the new beat thread establishes a new lock successfully.

**Comment by richardasaurus** [comment]
How do you kill it cleanly?

**Comment by pdoggi** [comment]
I'm figuring that out right now, but it may depend on how your setup is. I'm using [supervisor](https://github.com/Supervisor/supervisor) to run a celery beat daemon. My guess it has to do with the `stopsignal` in the conf file, but the [docs](http://supervisord.org/configuration.html?highlight=stopsignal) are unclear on what each type of signal means... 😞

I'll comment something when I learn more.

**Comment by pdoggi** [comment]
Okay, so I learned that [stop signals are basic kill signals in linux](https://unix.stackexchange.com/questions/317492/list-of-kill-signals). However, my setup with supervisor, I tried different stopsignals=`HUP`, `INT`, `QUIT`. All did not make a difference.

Instead, I decided to use your fix and let the redbeat lock timeout really fast so that the next respawned beat thread can establish a new lock. In the end, it seems like your solution works best @richardasaurus!
